File: 2017/ConstraintLayoutTransitions/ConstraintSetAnimation/app/src/main/java/me/mehdi/constraintsetanimation/MainActivity.java
package me.mehdi.constraintsetanimation;

import android.support.constraint.ConstraintLayout;
import android.support.constraint.ConstraintSet;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.transition.ChangeBounds;
import android.transition.Transition;
import android.transition.TransitionManager;
import android.view.View;
import android.view.animation.OvershootInterpolator;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    ConstraintLayout rootLayout;
    Button start;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rootLayout = findViewById(R.id.rootLayout);
        start = findViewById(R.id.start);

        start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ConstraintSet set = new ConstraintSet();
                set.clone(MainActivity.this, R.layout.second_keyframe);
                set.applyTo(rootLayout);
                Transition transition = new ChangeBounds();
                transition.setInterpolator(new OvershootInterpolator());
                transition.setDuration(2000);
                TransitionManager.beginDelayedTransition(rootLayout, transition);
            }
        });
    }

}

File: 2022/Transitions/app/src/main/java/com/example/playwithtransitions/MainActivity.kt
package com.example.playwithtransitions

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.animation.AnticipateInterpolator
import android.view.animation.AnticipateOvershootInterpolator
import android.view.animation.OvershootInterpolator
import androidx.constraintlayout.widget.ConstraintSet
import androidx.interpolator.view.animation.LinearOutSlowInInterpolator
import androidx.transition.AutoTransition
import androidx.transition.ChangeBounds
import androidx.transition.Fade
import androidx.transition.Scene
import androidx.transition.TransitionInflater
import androidx.transition.TransitionManager
import androidx.transition.TransitionSet
import androidx.transition.Visibility
import com.example.playwithtransitions.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding:ActivityMainBinding
    private lateinit var scene2: Scene
    private lateinit var mainScene: Scene
    private var transitionStarted = false
    private lateinit var myTransitionSet: TransitionSet


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)

        setContentView(binding.root)

        mainScene = Scene.getSceneForLayout(binding.root, R.layout.activity_main, this)
        scene2 = Scene.getSceneForLayout(binding.root, R.layout.other_layout, this)

        myTransitionSet = TransitionInflater.from(this)
            .inflateTransition(R.transition.my_transitions) as TransitionSet

        binding.hello.setOnClickListener{
//
//            TransitionManager.beginDelayedTransition(binding.root, myTransitionSet)
//            binding.group.visibility = View.VISIBLE
//            binding.hello.visibility = View.GONE


//            TransitionManager.go(scene2, myTransitionSet)
//            transitionStarted = true

            val constraintSet = ConstraintSet()
            constraintSet.clone(this, R.layout.other_layout)
            TransitionManager.beginDelayedTransition(binding.root, myTransitionSet)
            constraintSet.applyTo(binding.root)
        }
    }

    override fun onBackPressed() {
        if (transitionStarted){
            TransitionManager.go(mainScene)
            transitionStarted = false
        }
        else {
            super.onBackPressed()
        }
    }
}